### ISSUE TITLE
Return value is 0x double-padded in custom errors.

### ISSUE THREAD

**Issue by vittominacori** [NomicFoundation/hardhat#5793]
### Version of Hardhat

2.22.12

### What happened?

Hardhat returns 0x padded (twice) values in custom errors.

Note that it works with hardhat@2.22.11.

### Minimal reproduction steps

Having that sample contract:

```solidity
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.27;

contract SampleContract {
    error TestError(address account);

    constructor(address owner) {
        revert TestError(owner);
    }
}
```

and running the following test:

```ts
import hre from 'hardhat';

describe('SampleContract', function () {
  describe('Deployment', function () {
    it('Should fail', async function () {
      const [owner] = await hre.ethers.getSigners();
      const SampleContract = await hre.ethers.getContractFactory('SampleContract');
      await SampleContract.deploy(owner);
    });
  });
});
```

I receive the following error where the address is `0x` padded twice.

```
Error: VM Exception while processing transaction: reverted with custom error 'TestError("0x0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266")'
```

**Comment by Xanewok** [comment]
Hey @vittominacori, thanks a lot for the report!

This was a small mistake that we did when porting the tracing logic into our Ethereum Development Runtime (EDR) that we enabled by default in the latest [release](https://github.com/NomicFoundation/hardhat/releases/tag/hardhat%402.22.12).

I managed to reproduce the issue and can confirm that the fix in https://github.com/NomicFoundation/edr/pull/688 fixes the problem, so this should be fixed once we release a Hardhat version that uses the newest EDR that contains the fix.

cc @fvictorio
